src/player.cpp

```diff
@@ -10443,32 +10444,57 @@ bool player::can_wield( const item &it, bool interactive ) const
     return true;
 }
 
-bool player::wield(item* it, bool autodrop)
+bool player::wield(item* it, bool )
 {
     if (weapon.has_flag("NO_UNWIELD")) {
         add_msg(m_info, _("You cannot unwield your %s!  Withdraw them with 'p'."),
                 weapon.tname().c_str());
         return false;
     }
-    if (it == NULL || it->is_null()) {
-        if(weapon.is_null()) {
+
+    if( it == NULL || it->is_null() ) {
+        if( weapon.is_null() ) {
             return false;
         }
-        if (autodrop || volume_carried() + weapon.volume() <= volume_capacity()) {
-            moves -= item_handling_cost(weapon);
-            inv.add_item_keep_invlet(remove_weapon());
+
+        uimenu prompt;
+        prompt.text = string_format( _( "Stop wielding %s?" ), weapon.tname().c_str() );
+        std::vector<std::function<void()>> actions;
+
+        prompt.addentry( -1, volume_carried() + weapon.volume() <= volume_capacity(), '1', _( "Store in inventory" ) );
+        actions.push_back( [&]{
+            moves -= item_handling_cost( weapon );
+            inv.add_item_keep_invlet( remove_weapon() );
             inv.unsort();
-            recoil = MIN_RECOIL;
-            return true;
-        } else if (query_yn(_("No space in inventory for your %s.  Drop it?"),
-                   weapon.tname().c_str())) {
-            g->m.add_item_or_charges(posx(), posy(), remove_weapon());
-            recoil = MIN_RECOIL;
-            return true;
-        } else {
-            return false;
+        });
+
+        prompt.addentry( -1, true, '2', _( "Drop item" ) );
+        actions.push_back( [&]{ g->m.add_item_or_charges( posx(), posy(), remove_weapon() ); });
```

Comment:
Don't use 2D argument version of the functions unless necessary (for example, in mapgen). That `posx(), posy()` should be just `pos()`.